.github/workflows/pr-build.yaml

```diff
@@ -0,0 +1,381 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Build PR
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  LANG: en_US
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set Up Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Build Project
+        run: |
+          ./mvnw -V -B \
+            -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+            clean install
+      - name: Save Cache
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ github.sha }}
```

Comment:
This would make the cache single-use only, there's [an example](https://github.com/actions/cache/blob/master/examples.md#java---maven) on how to do it by hashing the POM files.

Reply:
if we hash the poms and we change the sources then we end up using wrong artifacts, isn't it ?

Reply:
GitHub actions provide a few ways to "solve" this:

- caching
- deploy artefacts

None of them is optimal but the caching one is the easier to implement as today and as further step, we could cache only what the project produces instead of the whole local maven repo as the cache has also some limitations in term of space.